src/java/com/zimbra/oauth/handlers/impl/GoogleContactsImport.java

```diff
@@ -378,6 +378,12 @@ public void importData(List<Integer> folderIds, boolean fullSync) throws Service
                         ZimbraLog.extensions.debug(
                             "Did not find 'connections' element in JSON response object. Response body: %s",
                             respContent);
+                        if (jsonResponse.has("error")) {
```

Comment:
Also, per the changes discussed for profile image sync (we will be handling 100 contacts at a time, creating them, then doing 100 more) consider the following scenario:

* Use has 500 contacts
* The first 300 contacts are fetched with no errors, and created
* The fourth fetch has an `error` in the response from google.

What is the expectation at this point? Delete the 300 imported contacts? Throw an exception and don't finish the rest? Complete the rest of the import ignoring the failed 100? Retry to fetch - if so, how many times?

Reply:
Also we may as well go ahead and add the scope to the required scopes constant so it isn't necessary to set.

Reply:
Can we set the sync token to indicate that 300 contacts have been fetched and then throw an exception. We can then sync the remaining 200 when we sync again.

Reply:
I will test later today if we can reuse a pageToken and for how long it remains valid. We should not rely on this as it may vary depending on the service (fb, google, etc).